value/vector.go

```diff
@@ -15,13 +15,33 @@ import (
 	"robpike.io/ivy/config"
 )
 
-type Vector []Value
+type Vector struct {
+	slice []Value
+}
+
+// Len returns the number of elements in v.
+func (v *Vector) Len() int { return len(v.slice) }
+
+// At returns the i'th element of v.
+func (v *Vector) At(i int) Value { return v.slice[i] }
+
+// Set sets v[i] = x.
+func (v *Vector) Set(i int, x Value) { v.slice[i] = x }
+
+// All returns all the elements in v, for reading.
+func (v *Vector) All() []Value { return v.slice[:len(v.slice):len(v.slice)] }
+
+// WriteAll returns all the elements in v, for writing.
+func (v *Vector) WriteAll() []Value { return v.slice }
```

Comment:
this name sounds like a verb, but it does no writing.
maybe AllWritable? something that reports the property rather than the hope.

Reply:
Changed to Writable. This method will be replaced in a followup with a better strategy for editing vectors.